#! /bin/sh /usr/share/dpatch/dpatch-run
## 50_octave-value-list-resize-2.9.dpatch by Rafael Laboissiere <rafael@debian.org>
##
## DP: Avoid crashes in the octave_value_list destructor.  This patch was
## DP: posted by J. W. Eaton in the help-octave mailing list.
## DP: See http://www.octave.org/mailing-lists/help-octave/2005/4266
## DP: (this patch only applies to octave2.9)

@DPATCH@

--- octave2.9-2.9.4.orig/src/oct-obj.h
+++ octave2.9-2.9.4/src/oct-obj.h
@@ -96,7 +96,20 @@
 
   void resize (octave_idx_type n) { data.resize (n); }
 
-  void resize (octave_idx_type n, const octave_value& val);
+  void resize (octave_idx_type n, const octave_value& val)
+  {
+    octave_idx_type len = length ();
+  
+    if (n > len)
+      {
+        data.resize (n);
+  
+        for (octave_idx_type i = len; i < n; i++)
+  	data[i] = val;
+      }
+    else if (n < len)
+      data.resize (n);
+  }
 
   octave_value_list& prepend (const octave_value& val);
 
--- octave2.9-2.9.4.orig/src/oct-obj.cc
+++ octave2.9-2.9.4/src/oct-obj.cc
@@ -43,22 +43,6 @@
   return true;
 }
 
-void
-octave_value_list::resize (octave_idx_type n, const octave_value& val)
-{
-  octave_idx_type len = length ();
-
-  if (n > len)
-    {
-      data.resize (n);
-
-      for (octave_idx_type i = len; i < n; i++)
-	data[i] = val;
-    }
-  else if (n < len)
-    data.resize (n);
-}
-
 octave_value_list&
 octave_value_list::prepend (const octave_value& val)
 {
